### PR TITLE
Binary Matrix 

### PR DESCRIPTION
**Pull request by soudabeh19**
(Option -b) Binary Matrix to represent the existence of difference in conditions pair according to the subject and file

### CODE REVIEW

File: verifyFiles.py [hunk 1]
```diff
@@ -135,10 +135,11 @@ def check_files(conditions_dict):
 # For instance:
 #  {'condition1 vs condition2': {'c/c.txt': 0, 'a.txt': 2}}
 #  means that 'c/c.txt' is identical for all subjects in conditions condition1 and condition2 while 'a.txt' differs in two subjects.
-def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,checksum_after_file_path,check_corruption,sqlite_db_path):
+def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,checksum_after_file_path,check_corruption,sqlite_db_path,flag):
```
Comment (glatard) [inline verifyFiles.py]: `flag` is not a good name for a parameter -> `return_detailed_differences`.

File: verifyFiles.py [hunk 2]
```diff
@@ -135,10 +135,11 @@ def check_files(conditions_dict):
...
     # For each pair of conditions C1 and C1
     product = ((i,j) for i in conditions_dict.keys() for j in conditions_dict.keys())
     diff={} # Will be the return value
+    bDiff={}
```
Comment (glatard) [inline verifyFiles.py]: A short comment would be appreciated.

File: verifyFiles.py (hunk 1 above)
Comment (glatard) [reply verifyFiles.py]: You should also document this in the function's comments.

File: verifyFiles.py [hunk 3]
```diff
@@ -169,38 +171,38 @@ def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_fro
...
 		log_info("Identified " + c +" and " + d +" as two different runs of the same condition")
 		is_intra_condition_run=True
-	        
-   
-            for file_name in path_names:
+	    for file_name in path_names:
                 diff[key][file_name]=0
+            for subject in conditions_dict[c].keys():
+		bDiff[key][subject]={}
+	    for file_name in path_names:
```
Comment (glatard) [inline verifyFiles.py]: I don't think you need to loop twice over `path_names`.

File: verifyFiles.py [hunk 4]
```diff
@@ -237,9 +239,10 @@ def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_fro
 			     dictionary_executables[file_name]=get_executable_details(conn,sqlite_db_path,file_name)
     if sqlite_db_path:
       conn.close()
-    return diff,metric_values,dictionary_executables
-
-
+    if flag:
```
Comment (glatard) [inline verifyFiles.py]: how about returning both anyways and removing the flag?

File: verifyFiles.py [hunk 5]
```diff
@@ -264,13 +267,13 @@ def get_executable_details(conn,sqlite_db_path,file_name):#TODO Intra condition
...
-        if file_name.endswith(metric['extension']):
+	 if file_name.endswith(metric['extension']):
             matching_metrics.append(metric)
     return matching_metrics
 
 # Executes the following command:
-#    'command condition1/subject_name/file_name condition2/subject_name/file_name'
-# and returns the stdout if and only if command was successful
+# 'command condition1/subject_name/file_name condition2/subject_name/file_name'
+# an returns the stdout if and only if command was successful
```
Comment (glatard) [inline verifyFiles.py]: there's a new typo here.

File: verifyFiles.py [hunk 6]
```diff
@@ -292,14 +295,14 @@ def read_checksum_from_file(checksums_after_file_path):
     return checksum_from_file_dict
 
 #Method get_conditions_checksum_dict creates a dictionary containing , the dictionaries under different condition with the condition as the key and subject
-#folder dictionaries as the value.
+#folder dictionairies as the value.
```
Comment (glatard) [inline verifyFiles.py]: new typo here too.

File: verifyFiles.py [hunk 7]
```diff
@@ -442,15 +494,19 @@ def main():
...
 	#and the file checksumFile,checkCorruption and the path to the sqlite file.
-        diff,metric_values,dictionary_executables=n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,args.checksumFile,args.checkCorruption,args.sqLiteFile)
-	if args.fileDiff is not None:
+       	if args.fileDiff is not None:
             log_info("Writing difference matrix to file "+args.fileDiff)
             diff_file = open(args.fileDiff,'w')
-	    diff_file.write(pretty_string(diff,conditions_dict))
-	    diff_file.close()
+	    Diff,metric_values,dictionary_executables=n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,args.checksumFile,args.checkCorruption,args.sqLiteFile)
+            diff_file.write(pretty_string(Diff,conditions_dict))
```
Comment (glatard) [inline verifyFiles.py]: don't capitalize variables (`Diff`).

**Review by glatard — CHANGES_REQUESTED** [review]
Nice work! Only minor changes are required, see inside comments.

File: verifyFiles.py (hunk 1 above)
Comment (soudabeh19) [reply verifyFiles.py]: sure

File: verifyFiles.py (hunk 3 above)
Comment (soudabeh19) [reply verifyFiles.py]: you are right

File: verifyFiles.py (hunk 2 above)
Comment (soudabeh19) [reply verifyFiles.py]: sure, I'll do

File: verifyFiles.py (hunk 4 above)
Comment (soudabeh19) [reply verifyFiles.py]: nice I'll do

File: verifyFiles.py (hunk 5 above)
Comment (soudabeh19) [reply verifyFiles.py]: Thanks for your detailed review

File: verifyFiles.py [hunk 8]
```diff
@@ -135,10 +135,12 @@ def check_files(conditions_dict):
 # For instance:
 #  {'condition1 vs condition2': {'c/c.txt': 0, 'a.txt': 2}}
 #  means that 'c/c.txt' is identical for all subjects in conditions condition1 and condition2 while 'a.txt' differs in two subjects.
-def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,checksum_after_file_path,check_corruption,sqlite_db_path):
+# return_detailed_differences parameter specifies which function-made dictionary should be returned according to the user's request
+def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,checksum_after_file_path,check_corruption,sqlite_db_path,return_detailed_differences):
```
Comment (glatard) [inline verifyFiles.py]: You don't need the last parameter now that you always return both matrices.

**Review by glatard — CHANGES_REQUESTED** [review]
Just a small remaining comment.

File: verifyFiles.py (hunk 8 above)
Comment (soudabeh19) [reply verifyFiles.py]: Oops! totally right

**Comment by glatard** [comment]
Hi @lalet, could you review this PR and make sure your stuff still works with these modifications?

**Review by glatard — APPROVED** [review]
Good to go! I'll just wait for @lalet's feedback before merging.

File: verifyFiles.py [hunk 9]
```diff
@@ -411,7 +458,8 @@ def main():
 	parser.add_argument("-k","--checkCorruption",help="If this flag is kept 'TRUE', it checks whether the file is corrupted")
 	parser.add_argument("-s","--sqLiteFile",help="The path to the sqlite file which is used as the reference file for identifying the processes which created the files")
 	parser.add_argument("-x","--execFile",help="Writes the executable details to a file")
-	args=parser.parse_args()
+	parser.add_argument("-b","--binaryMatrix",help="Matrix showes diferences according to the subject and file in the comparision of condition pairs")
```
Comment (lalet) [inline verifyFiles.py]: Could you please correct the typos in the help section of binaryMatix

**Review by lalet — CHANGES_REQUESTED** [review]
Good work @soudabeh19 !! 
I have requested for a few minor changes due to typos. 
If you have added a new parameter, I would recommend you to add it to the readme page as well.

File: verifyFiles.py (hunk 9 above)
Comment (soudabeh19) [reply verifyFiles.py]: Thanks for mentioning

**Review by lalet — APPROVED** [review]

**Pull request merged**